src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenSubmitApplication.java

```diff
@@ -81,9 +75,25 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
                 .state(Draft)
                 .build();
         }
+
+        State state;
+        if (caseDataCopy.getHelpWithFeesAppliedForFees().toBoolean()) {
+            state = AwaitingHWFDecision;
```

Comment:
do we have an integration to test AwaitingHWFDecision and  AwaitingPayment scenarios ?

Reply:
Should AwaitingHWFDecision.validate be called before moving to this state?

Reply:
Actually it looks like there is no validate for that state, so forget that

Reply:
Nitins Question: I did include a test and response.json file.

Shauns Question: yh there isnt, also i wasnt sure if i should add it as we have a event to validate this. Even AwaitingPayment.validate doesnt really validate anything relating to payment but checks casedata is valid.